Add Installation docs for JS Suite (#256)
smartdevicelink/sdl_app_library_guides: docs/Getting Started/Installation/index.md

Glassfish 5.0.0 only works on JDK 8 and lower.
!!!
!@

@![javascript]
The SDL JavaScript Suite comes with prebuilt bundles of the library. There is a vanilla JavaScript distribution of the library as well as one for Node.js. They are located in the lib/js/dist and lib/node/dist directories respectively. To use one of these bundles, include one of the following somewhere in your project:

>         const SDL = require('lib/js/SDL.min.js');

Reviewer: For the JS dist, it will likely be imported in the browser and so `require` wouldn't work, as that's a Node specific import syntax. Recommend clarifying that or changing the line to show how to do it in the browser: `<script src='lib/js/SDL.min.js'></script>`

OR

        const SDL = require('lib/node/SDL.min.js');

### Building The JavaScript Library    

> To build the library bundle from the source code, first clone the [SDL JavaScript Suite](https://github.com/smartdevicelink/sdl_javascript_suite) repository then, simply call:

Reviewer: A developer will need to set up git submodules so that the generator can reference the rpc_spec. Recommend trying to do a fresh clone of the JS suite and double checking that the build steps here will work

const server = app.listen(PORT, async function () {
    console.log('Server running on port', PORT);
});
```
This code will start up a server on port 3000 on localhost, and serve any files in the directory it is running in. Now that the server code is complete, create a new file named `index.html` that will contain the app logic and save it in the same directory. Make the contents of the HTML file the following:
```html
<html>
    <head>
        <script src='./SDL.min.js'></script>
    </head>
    <body>
        <script>
            console.log(SDL);
        </script>
    </body>
</html>
```
Finally, run the server by running this in the same directory: `node index.js`. The console should print `Server running on port 3000`. Go to your browser and enter `localhost:3000` in the address bar. If you open up your browser's console on the blank page that shows up you should see the SDL library version be printed and the imported SDL object. If so, then you have successfully set up SDL in your browser! If not, then make sure your SDL build name is correct, and that the HTML file, build file, and JS server file are all in the same directory.

## Node.js Setup

> This build allows you to create apps that run on your computer. [Node.js](https://nodejs.org/en/) and npm are the tools you will need to integrate this library into what will be your app.

Reviewer 2: ```suggestion
This build allows you to create apps that run through a server on your computer. You will need to have installed [Node.js](https://nodejs.org/en/) and npm before beginning work on this app.

```

> In a new directory, save your `SDL.min.js` build there, and then create a new file in the same directory named `index.js`. Make the contents of that file the following:

Reviewer 2: Same as above, how does a dev know where to get `SDL.min.js` or how to generate it?

> ```js

Reviewer 2: ```suggestion

```js
```

const SDL = require('./SDL.min.js');
console.log(SDL);
```

> Finally, run the file: `node index.js`. You should see the SDL library version be printed and the imported SDL object. If so, then you have successfully set up the SDL library!

Reviewer 2: ```suggestion

Finally, run the file: `node index.js`. You should see the SDL library version be printed and the imported SDL object. If so, then you have successfully set up the SDL library!
```

> be printed

Where?

## WebEngine App Setup

> WebEngine apps use the vanilla JavaScript build, and are set up in a similar fashion to JS apps where it will also run in the browser. Set up the `index.js` and `index.html` file like in the **Vanilla JavaScript Setup** section. The majority of the configuration for the app will now be separated into a `manifest.js` file and then imported into the `index.html` file. Create a `manifest.js` file like below and save it in the same directory as the other two files. The entrypoint's value should have the same name as your app's HTML file. 

Reviewer 2: ```suggestion
WebEngine apps use the vanilla JavaScript build, and are set up in a similar fashion to those JS apps because it will also run in a browser. Set up the `index.js` and `index.html` file like in the **Vanilla JavaScript Setup** section. The majority of the configuration for the app will now be separated into a `manifest.js` file and then imported into the `index.html` file. Create a `manifest.js` file like below and save it in the same directory as the other two files. The entrypoint's value should have the same name as your app's HTML file. 
```

> ```js

Reviewer 2: ```suggestion

```js
```

export default {
    "entrypoint": "./index.html",
    "appIcon": "./app_icon.png",
    "appId": "hello-webengine",
    "appName": "Hello WebEngine",
    "category": "DEFAULT",
    "additionalCategories": [],
    "locales": {
        "de_DE": {
            "appName": "Hallo JS",
            "appIcon": "./app_icon.png"
        }
    },
    "appVersion": "1.0.0",
    "minRpcVersion": "6.0",
    "minProtocolVersion": "5.0"
};
```

> Note that the manifest file is using the import/export module syntax; in the HTML file it should be imported in as a module:

Reviewer 2: ```suggestion

Note that the manifest file is using the import/export module syntax; in the HTML file it should be imported in as a module:

```

```js
import sdl_manifest from './manifest.js';
```

> Also note that in the transport configuration the parameters for `WebSocketClientConfig` will be empty. This is because for WebEngine apps those connection details are expected as query parameters in the URL. See below for an example of what the URL is expected to be once the server is running. `sdl-host` is the location of the SDL Core WebSocket server. `sdl-port` is the port of that WebSocket server. `sdl-transport-role` refers to SDL Core's role, which is as a server (as opposed to a client).

Reviewer 2: ```suggestion

In the transport configuration, the parameters for `WebSocketClientConfig` will be empty. For WebEngine apps those connection details are expected as query parameters in the URL. See below for an example of what the URL is expected to be once the server is running. `sdl-host` is the location of the SDL Core WebSocket server. `sdl-port` is the port of that WebSocket server. `sdl-transport-role` refers to SDL Core's role, which is as a server (as opposed to a client).

```

```
http://localhost:3000/?sdl-host=HOST&sdl-transport-role=ws-server&sdl-port=PORT
```

> ### Building The JavaScript Library    

Reviewer 2: I think this may answer my "where is the sdl.min.js" questions. Should it be up top? If devs are just supposed to be using releases, I don't think this section is necessary, and can just be in the README.

Author: Removing this section and clarifying the name of the build files at the start of the doc.

> To rebuild the library bundle from the source code, first clone the [SDL JavaScript Suite](https://github.com/smartdevicelink/sdl_javascript_suite) repository then, simply call:

Reviewer 2: ```suggestion
To rebuild the library bundle from the source code, first clone the [SDL JavaScript Suite](https://github.com/smartdevicelink/sdl_javascript_suite) repository then, simply call the following from within the sdl_javascript_suite directory and `SDL.min.js` files will be generated in the `libs/js/dist` and `lib/node/dist` folder.:
```

```
git submodule init
git submodule update
npm run build
```

> from within the sdl_javascript_suite directory and `SDL.min.js` files will be generated in the libs/js/dist and lib/node/dist folder.

Reviewer 2: ```suggestion
```

Glassfish 5.0.0 only works on JDK 8 and lower.
!!!
!@

@![javascript]

> The SDL JavaScript Suite comes with prebuilt bundles of the library in the form of `SDL.min.js` files. There is a vanilla JavaScript distribution of the library as well as one for Node.js. They are located in the lib/js/dist and lib/node/dist directories respectively.

Author: The release on Github just contains the sdl_javascript_suite repository at the time of the release. If they download that, the `SDL.min.js` file would still be in those `dist` folders. I'll include a link to the releases to make sure they're pulling the most recent release of the project and therefore, most recent builds.

## Vanilla JavaScript Setup
This build allows you to create apps that run on the browser. In order to have the built JS file be imported into your HTML, you'll need to run a simple web server that can serve that JS file. We will be using [Node.js](https://nodejs.org/en/) and [npm](https://docs.npmjs.com/about-npm/) for this task, but you can use any software that lets you serve HTML and JS to the browser.

> In a new directory, save your `SDL.min.js` build there, and then run the following: `npm init`. Press Enter repeatedly through the prompts to set up some default npm configuration. Then, install the `express` package by running `npm install express --save`. Express is a popular Node.js package that allows easy setup of a server. Then, create a `index.js` file in the same directory, and save the following to it:

Reviewer 2: ```suggestion

In a new directory, save your `SDL.min.js` file there, and then run the following: `npm init`. Press Enter repeatedly through the prompts to set up some default npm configuration. Then, install the `express` package by running `npm install express --save`. Express is a popular Node.js package that allows easy setup of a server. Then, create a `index.js` file in the same directory, and save the following to it:
```

```html
<html>
    <head>
        <script src='./SDL.min.js'></script>
    </head>
    <body>
        <script>
            console.log(SDL);
        </script>
    </body>
</html>
```

Finally, run the server by running this in a Terminal window in the same directory: `node index.js`. The console should print `Server running on port 3000`. Go to your browser and enter `localhost:3000` in the address bar. If you open up your browser's console on the blank page that shows up you should see the SDL library version be printed and the imported SDL object. If so, then you have successfully set up SDL in your browser! If not, then make sure your SDL build name is correct, and that the HTML file, build file, and JS server file are all in the same directory. If you still have  questions, ask them in the javascript-suite-help channel in the SDL Slack. You can sign up for our Slack [here](https://slack.smartdevicelink.com/).

## Node.js Setup
This build allows you to create apps that run through a server on your computer. You will need to have installed [Node.js](https://nodejs.org/en/) and [npm](https://docs.npmjs.com/about-npm/) are before beginning work on this app.

> In a new directory, save your `SDL.min.js` build there, and then create a new file in the same directory named `index.js`. Make the contents of that file the following:

Reviewer 2: ```suggestion

In a new directory, save your `SDL.min.js` file, then create a new file in the same directory named `index.js`. Make the contents of that file the following:
```
